File: benchmark/matrix/bench_matrix.c
#include <stdio.h>
#include <time.h>

int main(void) {
    struct timespec s, e;
    clock_gettime(CLOCK_MONOTONIC, &s);
    const int N = 200;
    int sum = 0;
    for (int i = 0; i < N; i++)
        for (int j = 0; j < N; j++)
            for (int k = 0; k < N; k++)
                sum += i * k;
    clock_gettime(CLOCK_MONOTONIC, &e);
    double ms = (e.tv_sec-s.tv_sec)*1000.0 + (e.tv_nsec-s.tv_nsec)/1e6;
    printf("result: %d  time: %.2fms\n", sum, ms);
    return 0;
}

File: benchmark/matrix/bench_matrix.rs
use std::time::Instant;

fn main() {
    let start = Instant::now();
    const N: i32 = 200;
    let mut sum: i32 = 0;
    for i in 0..N {
        for _j in 0..N {
            for k in 0..N {
                sum = sum.wrapping_add(i.wrapping_mul(k));
            }
        }
    }
    let ms = start.elapsed().as_secs_f64() * 1000.0;
    println!("result: {}  time: {:.2}ms", sum, ms);
}
